src/main/scala/translating/IRToBoogie.scala

```diff
@@ -102,6 +104,18 @@ class IRToBoogie(var program: Program, var spec: Specification) {
     List(relyProc, relyTransitive, relyReflexive)
   }
 
+  /**
+   * A predicate used to assert the value of all readonly memory.
+   * (Boogie does not like this it if it is too large due to it being a single expression)
+   *
+   * E.g.
+   *  val readOnlyMemoryFunction = readOnlyMemoryPredicate(memoryToCondition(program.readOnlyMemory), mem)
+   *  val readOnlyMemory = List(BFunctionCall(readOnlyMemoryFunction.name, List(mem), BoolBType))
+   */
+  private def readOnlyMemoryPredicate(readonly: List[BExpr], mem: BVar) : BFunction = {
+    BFunction("readonly_memory", List(BParam("mem", mem.bType)), BParam(BoolBType), Some(readonly.reduce((a, b) => BinaryBExpr(BoolAND, a, b))), List(externAttr))
+  }
+
```

Comment:
This isn't actually used is it? This approach was already tried and does not work

Reply:
Its not used

Reply:
Probably best to remove it then?